nilearn/decoding/decoder.py

```diff
@@ -106,41 +107,131 @@ def _check_param_grid(estimator, X, y, param_grid=None):
 
     """
     if param_grid is None:
+        param_grid = _default_param_grid(estimator, X, y)
+
+    else:
+        if isinstance(estimator, (RidgeCV, RidgeClassifierCV)):
+            param_grid = _wrap_param_grid(param_grid, "alphas")
+
+    return param_grid
+
+
+def _default_param_grid(estimator, X, y):
+    """Generate sensible default for param_grid.
+
+    Parameters
+    ----------
+    estimator: str, optional
+        The estimator to choose among:
+        %(classifier_options)s
+        %(regressor_options)s
+        Default 'svc'.
+
+    X: list of Niimg-like objects
+        See :ref:`extracting_data`.
+        Data on which model is to be fitted. If this is a list,
+        the affine is considered the same for all.
+
+    y: array or list of shape (n_samples)
+        The dependent variable (age, sex, IQ, yes/no, etc.).
+        Target variable to predict. Must have exactly as many elements as
+        3D images in niimg.
+
+    Returns
+    -------
+    param_grid: dict of str to sequence, or sequence of such. Sensible default
+    dict has size 1 for linear models.
+    """
+    param_grid = {}
+
+    # define loss function
+    if isinstance(estimator, LogisticRegression):
+        loss = "log"
+    elif isinstance(estimator, (LinearSVC, RidgeCV, RidgeClassifierCV, SVR)):
+        loss = "squared_hinge"
+    elif isinstance(estimator, (DummyClassifier, DummyRegressor)):
+        if estimator.strategy in ["constant"]:
+            message = (
+                "Dummy classification implemented only for strategies"
+                ' "most_frequent", "prior", "stratified"'
+            )
+            raise NotImplementedError(message)
+    else:
+        raise ValueError(
+            "Invalid estimator. The supported estimators are:"
+            f" {list(SUPPORTED_ESTIMATORS.keys())}"
+        )
+    # define sensible default for different types of estimators
+    if hasattr(estimator, "penalty") and (estimator.penalty == "l1"):
+        min_c = l1_min_c(X, y, loss=loss)
+    else:
+        min_c = 0.5
```

Comment:
What case does this correspond to ? Can you comment on this choice ?

Reply:
This was also part of the existing code, I don't know what the reasoning was. Should we open a new issue to discuss this?